<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="camera" default="/camera/color" />

    <arg name="cloud_topic" default="/os_cloud_node/points" />
    <arg name="info_topic" default="$(arg camera)/camera_info" />
    <arg name="camera_topic" default="$(arg camera)/image_raw" />
    <arg name="camera_frame" default="camera_color_optical_frame" />
    <arg name="cloud_frame" default="os_sensor" />
    <arg name="cloud_out_topic" default="$(arg cloud_topic)/rgb" />
    <arg name="image_out_topic" default="$(arg camera)/projected" />

    <!-- Nodelet manager for this pipeline -->
    <!-- <node
        pkg="nodelet"
        type="nodelet"
        args="manager"
        name="lidar_camera_manager"
        output="screen" /> -->

    <!-- <node
        pkg="image_proc"
        type="image_proc" 
        name="image_proc_node1" /> -->

    <!-- Run image_proc/rectify nodelet -->
    <node
        pkg="nodelet"
        type="nodelet"
        name="rectify_color"
        args="load image_proc/rectify lidar_camera_manager --no-bond" >
        
        <!-- Remap input topics -->
        <remap from="image_mono" to="$(arg camera)/image_raw" />
        <remap from="camera_info" to="$(arg camera)/camera_info" />

        <!-- Remap output topics -->
        <remap from="image_rect" to="$(arg camera)/image_rect_color" />
    </node>

    <node
        pkg="tf"
        type="static_transform_publisher"
        name="cloud_camera_tf"
        output="screen"
        args="-0.00706296 -0.11697807 -0.24420192 0.52218648 -1.52973857  0.97920602 $(arg camera_frame) $(arg cloud_frame) 50" />

    <node pkg="lidar_camera_calibration" type="lidar_camera_detection" name="lidar_camera_detection" output="screen">
        <param name="cloud_topic" value="$(arg cloud_topic)"/>
        <param name="camera_topic" value="$(arg camera_topic)"/>
        <param name="info_topic" value="$(arg info_topic)"/>
        <param name="cloud_out_topic" value="$(arg cloud_out_topic)"/>
        <param name="image_out_topic" value="$(arg image_out_topic)"/>

        <param name="camera_frame" value="$(arg camera_frame)"/>
        <param name="cloud_frame" value="$(arg cloud_frame)"/>
    </node>

    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find lidar_camera_calibration)/rviz/detection.rviz" />

    <node 
        name="camera_lidar_projection" 
        pkg="image_view"
        type="image_view"
        respawn="false"
        output="screen">
        
        <!-- Remap input topics -->
        <remap from="image" to="$(arg camera)/projected" />
    </node>

    <include file="$(find pycoral_ros)/launch/detect.launch">
    </include>

</launch>